bin/fuel-core/src/cli/run.rs

```diff
@@ -510,10 +510,45 @@ pub async fn get_service_with_shutdown_listeners(
     info!("Fuel Core version v{}", env!("CARGO_PKG_VERSION"));
     trace!("Initializing in TRACE mode.");
     // initialize the server
+    #[cfg(feature = "rocksdb")]
+    let mut combined_database =
+        CombinedDatabase::from_config(&config.combined_db_config)?;
+
+    #[cfg(not(feature = "rocksdb"))]
     let combined_database = CombinedDatabase::from_config(&config.combined_db_config)?;
 
     let mut shutdown_listener = ShutdownListener::spawn();
 
+    // TODO: Handle graceful shutdown of migration processes
+    #[cfg(feature = "rocksdb")]
+    let _migration_dbs_handles: Vec<_> = {
+        let on_chain_migration_handle = combined_database
+            .take_on_chain_rocksdb_handle()
+            .map(|rocks_db| tokio::task::spawn_blocking(|| migrate_db(rocks_db)));
+
+        let off_chain_migration_handle = combined_database
+            .take_off_chain_rocksdb_handle()
+            .map(|rocks_db| tokio::task::spawn_blocking(|| migrate_db(rocks_db)));
+
+        let relayer_migration_handle = combined_database
+            .take_relayer_rocksdb_handle()
+            .map(|rocks_db| tokio::task::spawn_blocking(|| migrate_db(rocks_db)));
+
+        let gas_price_migration_handle = combined_database
+            .take_gas_price_rocksdb_handle()
+            .map(|rocks_db| tokio::task::spawn_blocking(|| migrate_db(rocks_db)));
+
+        vec![
+            on_chain_migration_handle,
+            off_chain_migration_handle,
+            relayer_migration_handle,
+            gas_price_migration_handle,
+        ]
+        .into_iter()
+        .flatten()
+        .collect()
```

Comment:
These three calls seem redundant.

Reply:
Thanks, done in 7e55342be1 (and 5c70a379b2 to completely remove the join_handles)